main.cpp

```diff
@@ -138,6 +140,9 @@ class camera {
   }
 };
 
+//using a fixed timestep of 1/60(fps) for each frame
+constexpr std::chrono::nanoseconds timestep(16ms);	

```

Comment:
16ms is a little above 62 fps, 17ms is about 59 fps. 
This is why it's common to see 59hz displays. I would recommend using the standard 17ms.
```suggestion
constexpr std::chrono::nanoseconds timestep(17ms);	
```